### PR TITLE
Crc alrogithm handlers

### PR DESCRIPTION
**Pull request by kevin-gillespie**

### CODE REVIEW

File: src/target/target.c [hunk 1]
```diff
@@ -2546,31 +2544,7 @@ int target_checksum_memory(struct target *target, target_addr_t address, uint32_
 	}
```
Comment (kevin-gillespie) [inline src/target/target.c:2545]: Compare w/o the on-chip CRC algorithm crashes and fails with QSPI authentication failure.

File: src/target/target.c [hunk 2]
```diff
@@ -2546,31 +2544,7 @@ int target_checksum_memory(struct target *target, target_addr_t address, uint32_
...
 
 	retval = target->type->checksum_memory(target, address, size, &checksum);
-	if (retval != ERROR_OK) {
```
Comment (jdk-maxim) [inline src/target/target.c:2549]: Consider adding a Comment here to mention this section of code was removed, incase we might want to restore it later, or when doing merge compares in the future.

File: contrib/loaders/checksum/armv7m_crc.s [hunk 3]
```diff
@@ -34,6 +34,10 @@
 
 _start:
 main:
+	/* Setup VTOR to point to ISR vector */
+	ldr		r3, SCB
+	ldr		r2, =__isr_vector
+	str		r2, [r3, #8]
```
[the remark this answers is not in the document]
Comment (kevin-gillespie) [reply contrib/loaders/checksum/armv7m_crc.s:40]: Not currently inside of OpenOCD. I can add a comment here explaining the offset.

File: src/target/armv7m.c [hunk 4]
```diff
@@ -898,8 +898,9 @@ int armv7m_checksum_memory(struct target *target,
 
 	int timeout = 20000 * (1 + (count / (1024 * 1024)));
 
+    /* Algorithm starts at the base address, BKPT at offset 0x34 */
 	retval = target_run_algorithm(target, 0, NULL, 2, reg_params, crc_algorithm->address,
-			crc_algorithm->address + (sizeof(cortex_m_crc_code) - 6),
+			crc_algorithm->address + 0x34,
```
[the remark this answers is not in the document]
Comment (kevin-gillespie) [reply src/target/armv7m.c:903]: There isn't an easy way to do this. The legacy code just includes a binary blob and has the -6 offset without any comments. I can add more comments to this and the assembly file, but I don't think there is a good way to auto generate this offset.

File: src/target/target.c (hunk 2 above)
Comment (kevin-gillespie) [reply src/target/target.c:2549]: Ok will do.

**Review by jdk-maxim — APPROVED** [review]
Used the Windows artifact, verified new BP scheme works with the default remotetimeout of 2000ms.  Looks good to me.

**Pull request merged**